src/codeflare_sdk/utils/generate_yaml.py

```diff
@@ -622,7 +625,39 @@ def _create_oauth_sidecar_object(
     )
 
 
-def write_components(user_yaml: dict, output_file_name: str):
+def update_local_kueue_label(local_queue: str, namespace: str):
+    if local_queue is not None:
+        return local_queue
+    else:
+        try:
+            config_check()
+            api_instance = client.CustomObjectsApi(api_config_handler())
+            local_queues = api_instance.list_namespaced_custom_object(
+                group="kueue.x-k8s.io",
+                version="v1beta1",
+                namespace=namespace,
+                plural="localqueues",
+            )
+        except Exception as e:  # pragma: no cover
+            return _kube_api_error_handling(e)
+        for lq in local_queues["items"]:
+            if (
+                "annotations" in lq["metadata"]
+                and "kueue.x-k8s.io/default-queue" in lq["metadata"]["annotations"]
+                and lq["metadata"]["annotations"][
+                    "kueue.x-k8s.io/default-queue"
+                ].lower()
+                == "true"
+            ):
+                return lq["metadata"]["name"]
+        raise ValueError(
+            "Default Local Queue with kueue.x-k8s.io/default-queue: true annotation not found please create a default Local Queue or provide the local_queue name in Cluster Configuration"
+        )
```

Comment:
I think this should be it's own function to improve readability. Something like `get_default_queue` or something?